server/linter/rules/rules_entities.go

```diff
@@ -0,0 +1,27 @@
+package rules
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kubeshop/tracetest/server/linter/analyzer"
+	"github.com/kubeshop/tracetest/server/model"
+)
+
+type BaseRule struct {
+	id string
+}
+
+var _ Rule = &BaseRule{}
+
+func NewRule(id string) BaseRule {
+	return BaseRule{id}
+}
+
+func (r BaseRule) Id() string {
+	return r.id
+}
+
+func (r BaseRule) Evaluate(ctx context.Context, trace model.Trace, config analyzer.LinterRule) (analyzer.RuleResult, error) {
```

Comment:
Rules on the other hand, have very custom logic, which makes it more difficult to encapsulate in a single default method, for these, we should create an implementation of the `Rule` interface per rule

Reply:
I would say there's no need for this `BaseRule`. it can be replaced by a `func (r MyRule) ID() string {return someconstantID}` func, satisfying the interface. This also makes the ID immutable, and that's good in this context